src/wix/WixToolset.Core.Burn/Bundles/BurnCommon.cs

```diff
@@ -78,11 +78,14 @@ internal abstract class BurnCommon : IDisposable
         protected const UInt32 BURN_SECTION_OFFSET_FORMAT = 40;
         protected const UInt32 BURN_SECTION_OFFSET_COUNT = 44;
         protected const UInt32 BURN_SECTION_OFFSET_UXSIZE = 48;
-        protected const UInt32 BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE = 52;
-        protected const UInt32 BURN_SECTION_SIZE = BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE + 4; // last field + sizeof(DWORD)
+        protected const UInt32 BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE0 = 52;
 
         protected const UInt32 BURN_SECTION_MAGIC = 0x00f14300;
-        protected const UInt32 BURN_SECTION_VERSION = 0x00000002;
+        protected const UInt32 BURN_SECTION_VERSION = 0x00000003;
+        protected const UInt32 BURN_SECTION_COMPATIBLE_VERSION = 0x00000002;
+        protected const UInt32 BURN_SECTION_SIZE = 512;
+        protected const UInt32 BURN_SECTION_MAX_ATTACHEDCONTAINER_COUNT = (BURN_SECTION_SIZE - BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE0) / sizeof(UInt32);
```

Comment:
Investigate using the actual section size (`SizeOfRawData`) instead of using hardcoded values.